tests/test_factory.py

```diff
@@ -199,3 +199,16 @@ def test_create_poetry_fails_on_invalid_configuration():
   - 'description' is a required property
 """
     assert expected == str(e.value)
+
+
+def test_create_poetry_with_invalid_dev_dependencies():
```

Comment:
The test cases here should be focussed on the flag `with_dev` directly, ie. re-use exostomg fixtured to check if development dependencies were loaded and not loaded if `True` and `False` respecively.

Reply:
Tests added for that.